pkg/agent/flowexporter/exporter/exporter_test.go

```diff
@@ -21,147 +21,213 @@ import (
-		PrevReverseBytes:   0,
-	}
-	// Following consists of all elements that are in IANAInfoElements and AntreaInfoElements (globals)
-	// Need only element name and other are dummys
-	elemList := make([]*ipfixentities.InfoElementWithValue, len(IANAInfoElements)+len(IANAReverseInfoElements)+len(AntreaInfoElements))
-	for i, ie := range IANAInfoElements {
-		elemList[i] = ipfixentities.NewInfoElementWithValue(ipfixentities.NewInfoElement(ie, 0, 0, 0, 0), nil)
-	}
-	for i, ie := range IANAReverseInfoElements {
-		elemList[i+len(IANAInfoElements)] = ipfixentities.NewInfoElementWithValue(ipfixentities.NewInfoElement(ie, 0, 0, ipfixregistry.IANAReversedEnterpriseID, 0), nil)
-	}
-	for i, ie := range AntreaInfoElements {
-		elemList[i+len(IANAInfoElements)+len(IANAReverseInfoElements)] = ipfixentities.NewInfoElementWithValue(ipfixentities.NewInfoElement(ie, 0, 0, 0, 0), nil)
-	}
-
 	mockIPFIXExpProc := ipfixtest.NewMockIPFIXExportingProcess(ctrl)
 	mockDataSet := ipfixtest.NewMockIPFIXSet(ctrl)
 	mockIPFIXRegistry := ipfixtest.NewMockIPFIXRegistry(ctrl)
+
+	var recordv4, recordv6 flowexporter.FlowRecord
+	var elemListv4, elemListv6 []*ipfixentities.InfoElementWithValue
+	if v4Enabled {
+		recordv4 = getFlowRecord(false)
+		elemListv4 = getElemList(IANAInfoElementsIPv4, AntreaInfoElementsIPv4)
+	}
+	if v6Enabled {
+		recordv6 = getFlowRecord(true)
+		elemListv6 = getElemList(IANAInfoElementsIPv6, AntreaInfoElementsIPv6)
+	}
 	flowExp := &flowExporter{
 		nil,
 		mockIPFIXExpProc,
-		elemList,
+		elemListv4,
+		elemListv6,
 		testFlowExportFrequency,
 		0,
-		testTemplateID,
+		testTemplateIDv4,
+		testTemplateIDv6,
 		mockIPFIXRegistry,
+		v4Enabled,
+		v6Enabled,
 	}
 
-	for i, ie := range flowExp.elementsList {
+	// TODO: add tests for data fields
+	sendDataSet := func(elemList []*ipfixentities.InfoElementWithValue, templateID uint16, record flowexporter.FlowRecord) {
+		var dataSet ipfixentities.Set
+		mockDataSet.EXPECT().AddRecord(gomock.AssignableToTypeOf(elemList), templateID).DoAndReturn(
+			func(elements []*ipfixentities.InfoElementWithValue, templateID uint16) interface{} {
+				for i, ieWithValue := range elements {
+					assert.Equal(t, ieWithValue.Element.Name, elemList[i].Element.Name)
+					assert.Equal(t, ieWithValue.Value, elemList[i].Value)
+				}
+				return nil
+			},
+		)
+		mockDataSet.EXPECT().GetSet().Return(dataSet)
+		mockIPFIXExpProc.EXPECT().AddSetAndSendMsg(ipfixentities.Data, dataSet).Return(0, nil)
+
+		err := flowExp.sendDataSet(mockDataSet, record)
+		if err != nil {
+			t.Errorf("Error in sending data set: %v", err)
+		}
+	}
+
+	if v4Enabled {
+		sendDataSet(elemListv4, testTemplateIDv4, recordv4)
+	}
+	if v6Enabled {
+		sendDataSet(elemListv6, testTemplateIDv6, recordv6)
+	}
+}
+
+func getElemList(ianaIE []string, antreaIE []string) []*ipfixentities.InfoElementWithValue {
+	// Following consists of all elements that are in IANAInfoElements and AntreaInfoElements (globals)
+	// Need only element name and other are dummys
```

Comment:
s/other are dummys/other fields are set to dummy values

Reply:
Updated.